### PR TITLE
S3 migration command to migrate data from S3 to dStorage

### PR DESCRIPTION
**Pull request by iamrz1**

### CODE REVIEW

**Comment by guruhubb** [comment]
@iamrz1 do you have an updated readme and cli help for this? Please also do a UML flow of the architecture and detailed documentation

**Comment by iamrz1** [comment]
> @iamrz1 do you have an updated readme and cli help for this? Please also do a UML flow of the architecture and detailed documentation

Sure, on it.

**Comment by sculptex** [comment]
how do i build this to test? starting from scratch. I tried to pull then pull this branch but build number is same

**Comment by iamrz1** [comment]
> how do i build this to test? starting from scratch. I tried to pull then pull this branch but build number is same

Use the `streaming-upload` branch of go-sdk in go.mod file, then run `go mod tidy && go mod vendor` as usual. build binary with `make install`. Look up `S3 migration` in doc to migrate data from s3. It should work as expected.

**Comment by sculptex** [comment]
> > how do i build this to test? starting from scratch. I tried to pull then pull this branch but build number is same
> 
> Use the `streaming-upload` branch of go-sdk in go.mod file, then run `go mod tidy && go mod vendor` as usual. build binary with `make install`. Look up `S3 migration` in doc to migrate data from s3. It should work as expected.

okay, I think I built it,
./zbox version
Version info:
	zbox....:  v1.0.1-188-g2fc6997-dirty
	gosdk...:  v1.2.2


what docs are you referring to?

**Comment by iamrz1** [comment]
> okay, I think I built it

Great!

> what docs are you referring to?

Just the readme file in this PR

**Comment by sculptex** [comment]
okay I found in in the readme, unfortunately the hyperlink from the commands in the branch readme is hard coded to base zboxcli so it doesn't work.
Now that command is not in the zbox I built, I tried several methods including pulling that branch of gosdk and referencing it locally, none of them have that command.
There needs to be a reproduceable method for non-git-experts such as myself

File: cmd/s3-migration.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
+package cmd
```
Comment (stewartie4) [inline cmd/s3-migration.go:1]: Add unit tests for this file

File: config/migration.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package config
```
Comment (stewartie4) [inline config/migration.go:1]: Add unit tests

File: config/s3.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package config
```
Comment (stewartie4) [inline config/s3.go:1]: Add unit tests

File: service/status_bar.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package service
```
Comment (stewartie4) [inline service/status_bar.go:1]: Add unit tests

**Review by stewartie4 — CHANGES_REQUESTED** [review]
There seems to be missing unit test coverage

**Comment by Sriep** [comment]
Market place PRs https://github.com/0chain/gosdk/pull/34 and https://github.com/0chain/zboxcli/pull/23 have been merged, so recheck auth ticket use. Remove commented-out code and resolve todos and merge conflicts.

**Comment by cnlangzi** [comment]
@lpoli  should we close this PR?

**Pull request closed without merging**